pkg/properties/propertiesGet.go

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package properties
+
+import (
+	"context"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/galasa.dev/cli/pkg/api"
+	galasaErrors "github.com/galasa.dev/cli/pkg/errors"
+	"github.com/galasa.dev/cli/pkg/galasaapi"
+	"github.com/galasa.dev/cli/pkg/propertiesformatter"
+	"github.com/galasa.dev/cli/pkg/utils"
+)
+
+var (
+	validFormatters = CreateFormatters()
+)
+
+// GetProperties - performs all the logic to implement the `galasactl properties get` command,
+// but in a unit-testable manner.
+func GetProperties(
+	namespace string,
+	name string,
+	prefix string,
+	suffix string,
+	apiServerUrl string,
+	propertiesOutputFormat string,
+	console utils.Console,
+) error {
+	var err error
+	var chosenFormatter propertiesformatter.PropertyFormatter
+	
+	chosenFormatter, err = validateOutputFormatFlagValue(propertiesOutputFormat, validFormatters)
+	if err == nil {
+		var cpsProperty []galasaapi.CpsProperty
+		cpsProperty, err = getCpsPropertiesFromRestApi(namespace, name, prefix, suffix, apiServerUrl, console)
+		if err == nil {
+			var outputText string
+
+			//convert galasaapi.CpsProperty into formattable data
+			formattableProperty := FormattablePropertyFromGalasaApi(cpsProperty)
+			outputText, err = chosenFormatter.FormatProperties(formattableProperty)
+
+			if err == nil {
+				console.WriteString(outputText)
+			}
+
+		}
+	}
+
+	return err
+}
+
+// Retrieves properties from the ecosystem API that match a given namespace.
+// Multiple properties can be returned as the namespace is not unique.
+func getCpsPropertiesFromRestApi(
+	namespace string,
+	name string,
+	prefix string,
+	suffix string,
+	apiServerUrl string,
+	console utils.Console,
+) ([]galasaapi.CpsProperty, error) {
+
+	var err error = nil
+
+	var context context.Context = nil
+
+	// An HTTP client which can communicate with the api server in an ecosystem.
+	restClient := api.InitialiseAPI(apiServerUrl)
+
+	var httpResponse *http.Response
+	var cpsProperties = make([]galasaapi.CpsProperty, 0)
+
+	if name == "" {
+		apicall := restClient.ConfigurationPropertyStoreAPIApi.QueryCpsNamespaceProperties(context, namespace)
+		if prefix != "" {
+			apicall = apicall.Prefix(prefix)
+		}
+		if suffix != "" {
+			apicall = apicall.Suffix(suffix)
+		}
+		cpsProperties, httpResponse, err = apicall.Execute()
```

Comment:
This is in the if else block as the scope of the API call is limited to the if statement, since the calls are to two separate endpoints.
I had also made this suggestion during development, however I am unsure how to change the scope without impacting the functionality